posthog/models/special_migration.py

```diff
@@ -0,0 +1,43 @@
+from django.db import models
+
+
+# an enum, essentially
+class MigrationStatus:
+    NotStarted = 0
+    Running = 1
+    CompletedSuccessfully = 2
+    Errored = 3
+    RolledBack = 4
+
+
+class SpecialMigration(models.Model):
+    class Meta:
+        constraints = [models.UniqueConstraint(fields=["name"], name="unique name",)]
+
+    id: models.BigAutoField = models.BigAutoField(primary_key=True)
+    name: models.CharField = models.CharField(max_length=400, null=False, blank=False)
+    progress: models.IntegerField = models.IntegerField(null=False, blank=False, default=0)
+    status: models.PositiveSmallIntegerField = models.PositiveSmallIntegerField(
+        null=False, blank=False, default=MigrationStatus.NotStarted
+    )
+
+    current_operation_index: models.PositiveSmallIntegerField = models.PositiveSmallIntegerField(
+        null=False, blank=False, default=0
+    )
+    current_query_id: models.CharField = models.CharField(max_length=400, null=False, blank=False, default="")
+    celery_task_id: models.CharField = models.CharField(max_length=400, null=False, blank=False, default="")
+
+    started_at: models.DateTimeField = models.DateTimeField(null=True, blank=True)
+
+    # Can finish with status 'CompletedSuccessfully', 'Errored', or 'RolledBack'
+    finished_at: models.DateTimeField = models.DateTimeField(null=True, blank=True)
+
+    error: models.TextField = models.TextField(null=True, blank=True)
```

Comment:
this should be `last_error` if we are thinking to update it without appending

Reply:
Well at least the way I'm building this, there will only ever be one error to deal with.

Migration starts, if it errors, we bail and show the error. If we restart, this is cleared. Indeed, I can see how it is the last error but given that you only get a new error if you re-run, the distinction seems clear to me whereas `last_error` might actually be confusing?

Reply:
Sure but I think what you described is an implementation detail: when the migration starts you already update a bunch of columns, you could technically blank this as well to archive the consistency you were mentioning.

Reply:
Updated to `last_error`